File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/domain/events/GiftCardCredited.java
package it.giannibombelli.wsc2026.giftcard.domain.events;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

public record GiftCardCredited(
    GiftCardId aggregateId,
    Money creditedAmount
) implements GiftCardEvent {
    public GiftCardCredited {
        Require.requireArgument(aggregateId, "giftCardId");
        Require.requireArgument(creditedAmount, "creditedAmount");
    }
}

File: typescript/src/giftcard/domain/events/giftCardCredited.ts
// Evento emesso quando una gift card viene accreditata.

import type { Event } from '@/common/domain/model/event.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { GiftCardId } from '../giftcard/giftCardId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type GiftCardCredited = Event<'GiftCardCredited', GiftCardId> & {
  readonly creditedAmount: Money;
};

export function giftCardCredited(aggregateId: GiftCardId, creditedAmount: Money): GiftCardCredited {
  requireArgument(aggregateId, 'giftCardId');
  requireArgument(creditedAmount, 'creditedAmount');
  return { kind: 'GiftCardCredited', aggregateId, creditedAmount };
}
